server/lomas_server/dp_queries/dp_libraries/smartnoise_synth.py

```diff
@@ -291,18 +345,22 @@ def _model_pipeline(
                 raise InvalidQueryException(
                     "Error while selecting provided select_cols: " + str(e)
                 ) from e
+        if query_json.synth_name == SSynthMarginalSynthesizer.MWEM:
+            if private_data.shape[1] > 3:
+                raise InvalidQueryException(  # TODO improve by looking better
+                    "MWEMSynthesizer does not allow too high cardinality. Select "
```

Comment:
any reason why this synthesizer does not allow high level of cardinalities ?

Reply:
In theory, no reason. In practice, it makes the server crash without any error message because it stores everything in memory.

I want to discuss what we should do at the meeting today and improve later, 3 columns is very arbitrary

Reply:
especially as it can be binned continuous columns versus just boolean columns